consul/consul
# Poll Shift and OfficerAssignment fixes

Where
=====
Current Poll effort

What
====
- Add Shift task to composed index on its table
- Remove composed index from OfficerAssignment preventing "temporary" relationship with Shift from working on multiple shifts per date and officer
- After creating a Shift with a recount&scrutiny task, create the associated OfficerAssignments with the `final` flag active.
- Before destroying a Shift, destroy as well all related OfficerAssigments that could have been created because of it.

Screenshots
===========
## Two Shifts with the same date/booth/officer but different tasks 
![screen shot 2017-10-03 at 12 21 27](https://user-images.githubusercontent.com/983242/31123162-06a19b32-a83f-11e7-9a6d-6fddf41c37ca.jpg)

Test
====
- Check OfficerAssigments are created associated with correct `final` flag value for each Shift
- Check when destroying a Shift only associated OfficerAssigments are destroyed
- Check on a feature spec that user can create a Shift for each task type on the same date/officer/booth 

Deployment
==========
- As usual

Warnings
========
- Pending to remove many OfficerAssignment usage internally, but not the right time to do so
